dvc/dependency/repo.py

```diff
@@ -67,7 +67,7 @@ def _get_checksum(self, locked=True):
                     return self.repo.cache.local.tree.get_hash(path, tree=tree)
                 return tree.get_file_hash(path)
 
-    def status(self):
+    def workspace_status(self):
         current_checksum = self._get_checksum(locked=True)
         updated_checksum = self._get_checksum(locked=False)
 
```

Comment:
This status hack backfired a bit. Need to set self.tree properly here (it will pretty much be RepoTree for the external repo that this belongs to), so that things like `self.exists()` work automatically.